01-shapefile-attributes.Rmd

```diff
@@ -75,182 +79,292 @@ CSV to Shapefile in R</a>
+for each line (or point/ polygon if using those) in the shapefile. The 0-12
+printed in the first column is simply a `row.name`. We can test this by issuing
+the following command to the `R` console: `row.names(aoiBoundary@data)`.
+
+``` {r row-names}
+#What are the row names?
+row.names(lines_HARV@data)
+```
+
+If we want to select something about the attribuates, we can do this by 
+combining common functions (e.g. `names` or `head`) and then using `NAME@data`.
 
 Let's give it a try.
 
 ```{r view-shapefile-attributes }
-
 #view just the attribute names for the lines spatial object
-names(lines@data)
-
-#view all attribute data for the lines spatial object
-lines@data 
+names(lines_HARV@data)
 
 #view just the top 5 attributes for the lines spatial object
-head(lines@data)
+head(lines_HARV@data)
 
-#view attributes for the other spatial objects
-aoiBoundary@data
-point@data
+```
+
+Alternatively, you can find attributes by using the `attributes` function,
+although if you have lots of objects in your shapefile this can be difficult to
+look through for the information you need.  
+
+```{r view shapefile-attributes-2}
+attributes(lines_HARV)
 
 ```
 
-#CHALLENGE
-Explore the attributes associated with the `point` and `aoiBoundary` spatial objects. 
+#Challenge: Attributes in Different Spatial Classes
+Explore the attributes associated with the `point_HARV` and `aoiBoundary_HARV` spatial objects. 
 
 1. How many attributes do each have?
-2. Who owns the site in the `point` data object?
+2. Who owns the site in the `point_HARV` data object?
 3. Which of the following are NOT attributes of the `point` data object?
     A) Lat B) County C) Country
 
-#Exploring Values in One Attribute
-We can also explore individual values stored within a particular attribute. Again,
-comparing attributes to a spreadsheet or a `data.frame`, this is similar to 
-exploring values in a column. We can do this using the `$`.
+```{r challenge-code-attributes-classes, results="hide", echo=FALSE}
+#1
+names(point_HARV@data)  #14 attributes
+names(aoiBoundary_HARV@data)  #1 attribute
+
+#2
+head(point_HARV@data)  #Harvard University, LTER
+
+#3
+point_HARV@data  # C Country
+```
+
+
+#Explore Values in One Attribute
+We can also explore individual values stored within a particular attribute.
+Again, comparing attributes to a spreadsheet or a `data.frame`, this is similar
+to exploring values in a column. We can do this using the `$` and the name of
+the attribute. 
 
 ```{r explore-attribute-values }
 #view all attributes in the lines shapefile within the TYPE field
-lines$TYPE
+lines_HARV$TYPE
 
-#view unique attributes in the lines shapefiles within the Type field
-levels(lines$TYPE)
+#view unique attributes in the lines shapefiles within the TYPE field
+levels(lines_HARV$TYPE)
 
 ```
 
-#Acessing levels for categorical variables
+Levels is particularly useful if we have a object with lost of different lines,
```

Comment:
?? please check spelling and edit before submitting a PR.

I'd just remove this sentence however given we have a header below that works well.